### PR TITLE
[PTQ][OV] New quantization estimators

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

1) New reducers and aggregators was added for OV backend:

- MeanReduce
- QuantileReduce
- AbsQuantileReduce
- MedianAggregator
- MeanNoOutliersAggregator
- MedianNoOutliersAggregator

2) 

- Reducers now can handle several inputs / several outputs
- Experimental TensorCollector now can handle reducers with several inputs/outputs

3) Bias correction and Fast bias correction statistics aggregator tests are presented

### Reason for changes

To extend available TensorCollectors to be alligned with POT

### Related tickets

102715

### Tests

- Unit tests for all OV reducers/aggregators in `tests/experimental/common/test_reducers_and_aggregators.py`
- `tests/experimental/common/test_statistic_collector.py` updated
- Inplace OV mean reducer tested in `tests/openvino/native/test_model_transformer.py`

There are no degradation in `test_quantize_conformace.py` test

### CODE REVIEW

**Comment by ksilligan** [comment]
run pytorch pre-commit tests

File: nncf/common/tensor_statistics/collectors.py [hunk 1]
```diff
@@ -116,23 +117,27 @@ class NNCFCollectorTensorProcessor(ABC):
 
     @staticmethod
     @abstractmethod
-    def reduce_min(x: NNCFTensor, axis: Union[int, tuple, list]) -> NNCFTensor:
+    def reduce_min(x: NNCFTensor, axis: Union[int, tuple, list], keepdims: bool = False) -> NNCFTensor:
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply nncf/common/tensor_statistics/collectors.py:120]: Good catch! But I suggest to keep this param because `axis` is standard name in numpy and even tensorflow. Torch uses singular form for `dim` as well

File: tests/experimental/common/test_reducers_and_aggregators.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
+import pytest
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/experimental/common/test_reducers_and_aggregators.py:14]: Thank you! Done

File: nncf/common/tensor_statistics/collectors.py [hunk 3]
```diff
@@ -211,6 +261,48 @@ def sum(tensor: NNCFTensor) -> TensorElementsType:
...
+    @staticmethod
+    @abstractmethod
+    def quantile(tensor: NNCFTensor, quantile: Union[float, List[float]],
+                 axis: Union[int, tuple, list]) -> List[TensorElementsType]:
+        """
+        Compute the quantile-th percentile(s) of the data along the specified axis.
+
+        :param tensor: Given NNCFTensor.
+        :params quantile: Percentile or sequence of percentiles to compute, which must be between
+            0 and 1. inclusive.
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply nncf/common/tensor_statistics/collectors.py]: Done

**Review by alexsu52 — APPROVED** [review]
LGTM

**Review by l-bat — APPROVED** [review]
